examples/model_selection/RF_vs_SPORF_random_search.py

```diff
@@ -0,0 +1,221 @@
+An important step in classifier performance comparison is hyperparameter 
+optimization. Here, we specify the classifer models we want to tune and a 
+dictionary of hyperparameter ranges (preferably similar for fairness in 
+comparision) for each classifier. Then, we find the optimal hyperparameters 
+through a function that uses RandomizedSearchCV and refit the optimized 
+models to obtain accuracies. We can see clearly in the plot that the 
+optimized models perform better than the default parameter models. On the 
+dataset we use in this example, car dataset from OpenML-CC18, SPORF also 
+performs better than RF overall. 
+"""
+print(__doc__)
+
+from sklearn.model_selection import RandomizedSearchCV
+from sklearn.model_selection import GridSearchCV
+
+import numpy as np
+import math
+from rerf.rerfClassifier import rerfClassifier
+from sklearn.ensemble import RandomForestClassifier
+import openml
+from sklearn.model_selection import train_test_split
+from sklearn import metrics
+from warnings import simplefilter
+
+simplefilter(action="ignore", category=FutureWarning)
+from warnings import simplefilter
+
+simplefilter(action="ignore", category=FutureWarning)
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+
+
+def hyperparameter_optimization_random(X, y, *argv):
+    """
+    Given a classifier and a dictionary of hyperparameters, find optimal hyperparameters using RandomizedSearchCV.
+
+    Parameters
+    ----------
+    X : numpy.ndarray
+        Input data, shape (n_samples, n_features)
+    y : numpy.ndarray
+        Output data, shape (n_samples, n_outputs)
+    *argv : list of tuples (classifier, hyperparameters)
+        List of (classifier, hyperparameters) tuples:
+
+        classifier : sklearn-compliant classifier
+            For example sklearn.ensemble.RandomForestRegressor, rerf.rerfClassifier, etc
+        hyperparameters : dictionary of hyperparameter ranges
+            See https://scikit-learn.org/stable/modules/generated/sklearn.model_selection.RandomizedSearchCV.html.
+
+    Returns
+    -------
+    clf_best_params : dictionary
+        Dictionary of best hyperparameters
+    """
+
+    clf_best_params = {}
+
+    # Iterate over all (classifier, hyperparameters) pairs
+    for clf, params in argv:
+
+        # Run randomized search
+        n_iter_search = 10
+        random_search = RandomizedSearchCV(
+            clf, param_distributions=params, n_iter=n_iter_search, cv=10, iid=False
+        )
+        random_search.fit(X, y)
+
+        # Save results
+        clf_best_params[clf] = random_search.best_params_
+
+    return clf_best_params
+
+
+###############################################################################
+# Building classifiers and specifying parameter ranges to sample from
+# ----------------------------------------------------------
+#
+
+# get some data
+task_id = 146821  # car
+openml.config.apikey = "c9ea8896542dd998ea42685f14e2bc14"
```

Comment:
Not sure if this is supposed to be public. When dealing with keys, you can save it to a different file and just read off the file within a script.